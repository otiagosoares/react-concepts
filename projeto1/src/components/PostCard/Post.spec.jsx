import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { PostCard } from '.';

const mock = {
  id: 1,
  title: 'title post 1',
  body: 'body text',
  cover: 'img/img.png',
};

describe('<PostCard />', () => {
  it('should rendered PostCard on screed correctly', () => {
    render(<PostCard {...mock} />);

    const image = screen.getByRole('img', { name: mock.title });
    const heading = screen.getByRole('heading', { name: 'title post 1 - 1' });

    expect(image).toHaveAttribute('src', mock.cover);
    expect(heading).toBeInTheDocument();
  });

  it('should match snapchot', () => {
    const { container } = render(<PostCard {...mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
