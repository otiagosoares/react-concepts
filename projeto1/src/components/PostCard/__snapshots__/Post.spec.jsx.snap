// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapchot 1`] = `
<article
  class="post"
>
  <div
    class="post-thumb"
  >
    <img
      alt="title post 1"
      src="img/img.png"
    />
  </div>
  <div
    class="post-content"
  >
    <h2>
      title post 1
       - 
      1
    </h2>
    <p>
      body text
    </p>
  </div>
</article>
`;
